docs/java/how-to-set-a-schedule-to-start-timeout-in-java.md

```diff
@@ -0,0 +1,41 @@
+---
+id: how-to-set-a-schedule-to-start-timeout-in-java
+title: How to set a Schedule-To-Start Timeout in Java
+sidebar_label: Schedule-To-Start Timeout
+description: To set a Schedule-To-Start Timeout, use `ActivityOptions.newBuilder.setScheduleToStartTimeout​`].
+tags:
+  - Java
+  - how-to
+  - developer-guide
+---
+
+To set a [Schedule-To-Start Timeout](/docs/concepts/what-is-a-schedule-to-start-timeout), use [`ActivityOptions.newBuilder.setScheduleToStartTimeout​`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/activity/ActivityOptions.Builder.html).
+
+- Type: `Duration`
+- Default: Unlimited. This timeout is non-retryable.
+
+- With `ActivityStub`
```

Comment:
What is the difference between with `ActivityStub` and with `WorkflowImplementationOptions` ?

Reply:
`ActivityStub` is implemented within a Workflow implementation and `WorkflowImplementationOptions` are implemented at the Worker level.

Reply:
I debated calling these 'Setting options within a Workflow' and 'Setting options within a Worker' but at the `ActivityOptions` reference level this made sense.

Reply:
So, the consequences of this are that - you could have a conflict between options?

I assume that Activity Options set on the Worker would apply to all Activities executed by that Worker?
What if a Workflow sets Activity Options that conflict of those set at the Worker?

Reply:
The `ActivityOptions` set within a Workflow always take precedence. 
This is called out in the "How to set Activity Options in Java"; > this content however has no place in the app-dev guide; I can break it up so that the Worker related stuff goes into Worker creation, and the other bit goes into either Activity Definition or Workflow Definition.

The examples within options are included to show the two different formats, though the Options themselves are the same.

Reply:
Added:
"You can set Activity Options using an `ActivityStub` within a Workflow implementation, or per-Activity using `WorkflowImplementationOptions` within a Worker.
Note that if you define options per-Activity Type options with `WorkflowImplementationOptions.setActivityOptions()`, setting them again specifically with `ActivityStub` in a Workflow will override this setting."
To each of the Activity Options How-tos.